lightning/restore/checksum.go

```diff
@@ -84,7 +85,11 @@ func newChecksumManager(rc *RestoreController) (ChecksumManager, error) {
 
 		manager = newTiKVChecksumManager(store.(tikv.Storage).GetClient(), pdCli)
 	} else {
-		manager = newTiDBChecksumExecutor(rc.tidbMgr.db)
+		e, ok := rc.tidbGlue.GetSQLExecutor().(glue.ExternalTiDBGlue)
```

Comment:
should TiDBChecksumExecutor use Glue directly here directly? Why we need to care about the implementation of glue here?

Reply:
as below error message, if a TiDB (with lightning) runs with PD which version is less than v4.0.0, this function will call `newTiDBChecksumExecutor` which need a `*sql.DB` that only `ExternalTiDBGlue` owns. I want to give a friendly message about this scenario